### PR TITLE
common: fixing warnings on windows

### PR DESCRIPTION
**Pull request by mgrudzinska**
cstring was not included

### CODE REVIEW

File: src/lib/tvgInitializer.cpp [hunk 1]
```diff
@@ -43,25 +43,31 @@ static uint16_t _version = 0;
...
-    auto p = SRC;
     const char* x;
+    auto p = SRC;
+
+    x = p;
+    while (*x != '\0') ++x;
+    auto len = x - SRC;
 
     char major[3];
-    x = strchr(p, '.');
```
Comment (mgrudzinska) [inline src/lib/tvgInitializer.cpp:50]: I could include string.h instead of these changes.. should I?

**Comment by mgrudzinska** [comment]
#839

File: src/loaders/svg/tvgSvgLoaderCommon.h [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 #include "tvgCommon.h"
 #include "tvgArray.h"
+#include <string.h>
```
Comment (JSUYA) [inline src/loaders/svg/tvgSvgLoaderCommon.h:27]: According to your pr, we don't use str functions anymore, do we need this header?

File: src/loaders/svg/tvgSvgUtil.cpp [hunk 3]
```diff
@@ -263,7 +263,9 @@ string svgUtilURLDecode(const char *src)
 {
     if (!src) return nullptr;
 
-    auto length = strlen(src);
+    const char* x = src;
```
Comment (JSUYA) [inline src/loaders/svg/tvgSvgUtil.cpp:266]: How about `auto` instead of const char*?

File: src/loaders/svg/tvgSvgUtil.cpp [hunk 4]
```diff
@@ -290,7 +292,10 @@ string svgUtilBase64Decode(const char *src)
...
```
Comment (JSUYA) [inline src/loaders/svg/tvgSvgUtil.cpp:295]: How about `auto` instead of const char*?

File: src/loaders/svg/tvgSvgLoaderCommon.h (hunk 2 above)
Comment (mgrudzinska) [reply src/loaders/svg/tvgSvgLoaderCommon.h:27]: we are using functions from cstring in two places. 
One is the 'src/lib/tvgInitializer.cpp' and I changed a bit the code, so we don;t need to include 'cstring'
The other one is svg loader - in all files. also all files but 'tvgSvgUtil' have 'tvgSvgLoaderCommon' included. So I've included 'cstring' in the common file, and I've changed 'tvgSvgUtil' so it doesn't need 'cstring' any more

Comment (mgrudzinska) [reply src/loaders/svg/tvgSvgLoaderCommon.h:27]: @JSUYA but I included `string.h` no matter on what platform we compile the lib. I can't find any info why it's necessary only on windows...

File: src/loaders/svg/tvgSvgUtil.cpp (hunk 3 above)
Comment (mgrudzinska) [reply src/loaders/svg/tvgSvgUtil.cpp:266]: sorry, I forgot about that... changed

File: src/loaders/svg/tvgSvgUtil.cpp (hunk 4 above)
Comment (mgrudzinska) [reply src/loaders/svg/tvgSvgUtil.cpp:295]: sorry, I forgot about that... changed

**Review by hermet — CHANGES_REQUESTED** [review]
Why not including string...? I think there's no much gain, but code complexity is increased.
Any specific reason?

**Comment by JSUYA** [comment]
> Why not including string...? I think there's no much gain, but code complexity is increased. Any specific reason?

I was concerned that problems related to str function would continue to occur depending on the build environment.
So I thought not using it might be a better solution.
I did some tests and modified the github action script and build environment appropriately. (https://github.com/Samsung/thorvg/pull/839)

@mgrudzinska I am sorry to bother you.

**Comment by mgrudzinska** [comment]
one more time - #839 :)

**Pull request closed without merging**